BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -102,9 +106,33 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     public func start() async {
         guard !started else { return }
         started = true
-        if await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
-                                              defaultValue: false) {
-            subscribeToAppState()
+
+        guard await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
+                                                 defaultValue: false) else {
+            return
+        }
+
+        syncSettingSubscriberTask = Task {
+            for await (userId, _) in await self.stateService.syncToAuthenticatorPublisher().values {
```

Comment:
🤔 Isn't `shouldSync` (2nd parameter) needed anymore?

Reply:
It's not needed currently because we re-check that in `determineSyncForUserId(userId)`. The key piece of info we're getting from this subscription is that this user's sync setting has been _updated_ and therefore we need to re-evaluate. Since we re-evaluate based on _both_ vault and sync status, we don't need to pass this in here and we can just look it up.

That said, I suppose we could optionally pass this in to save ourselves that extra lookup. But it might make things a bit complicated in `determineSyncForUserId(userId)` (because you could be landing there with a vault update instead, where we would have to look this one up)